src/pages/core/advanced/measuring-time.mdx

```diff
@@ -0,0 +1,32 @@
+---
+tags: ["core", "advanced"]
+---
+
+import { Callout } from "nextra/components";
+
+# Measuring time
+
+Accessing the current time is useful in a lot of different contexts but time is
+hard. It is hard to coordinate, hard to keep in sync, and this gets worse in
+distributed settings such as blockchains.
+
+In CosmWasm, we solve this by passing you some information about the blockchain
+you are running on whenever en entrypoint is invoked.
+
+In each of the entrypoints, you get a parameter of the type `Env` and this
+struct contains the field `block`. The struct contained in this field has a
+bunch of different information about the current state of the blockchain you are
+running on.
+
+> The documentation about the `BlockInfo` struct can be
+> [found here](https://docs.rs/cosmwasm-std/2.0.4/cosmwasm_std/struct.BlockInfo.html)
+
+The timestamp contained in this struct can be safely used in your program as the
+source of the current time. Well, kinda. It won't be 100% matching the current
+time, as it will only definitely match the time the latest block on the
```

Comment:
fwiw, I _think_ this is the "time" contracts see... but this is new research for me personally, someone else may have more hands-on experience at the chain level here:

> The timestamp can be computed using [PBTS](https://docs.cometbft.com/main/spec/consensus/proposer-based-timestamp/index.html) or [BFT Time](https://docs.cometbft.com/main/spec/consensus/bft-time) algorithms. In case of PBTS, it is the time at which the proposer has produced the block (the value of its local clock). In case of BFT Time, it is equal to the weighted median of timestamps present in the previous commit.

- https://docs.cometbft.com/main/spec/core/data_structures#header

Reply:
"latest" is a bit ambiguous. I think this is the time of the block that is being executed. As David mentioned already, it comes from the cometBFT block header. According to the cometBFT docs, PBTS was only introduced in cometBFT 1.0, so the chains are currently using BFT Time (I wouldn't go that much into detail in our docs though).

Reply:
I changed the wording. Hope that the new wording is at least a little better.